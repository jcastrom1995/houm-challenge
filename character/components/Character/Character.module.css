.character {
  border: 1px solid var(--colors-gray-300);
  border-radius: var(--border-radius-semi);
  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.1), 0px 1px 2px rgba(0, 0, 0, 0.06);
  cursor: pointer;
}

.about {
  border-top: 1px solid var(--colors-gray-300);
  padding: 0.75rem;
}

.information {
  align-items: center;
  display: flex;
  justify-content: space-between;
  text-align: right;
}

.information svg {
  color: var(--color-icon);
}

.photo {
  display: block;
  position: relative;
  width: 100%;
  height: auto;
}

.image {
  display: block;
  border-radius: var(--border-radius-semi) var(--border-radius-semi) 0 0;
}
