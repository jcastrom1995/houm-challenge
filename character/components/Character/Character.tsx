import { ReactElement } from "react";
import Image from "next/image";
import { Text, Field } from "ui/components";
import * as Icon from "Icons";
import s from "./Character.module.css";

interface CharacterI {
  photo: string;
  name: string;
  gender: string;
  location: string;
  onSelect: () => void;
}

const Character = ({
  photo,
  name,
  gender,
  location,
  onSelect,
}: CharacterI): ReactElement => {
  return (
    <article onClick={onSelect} className={s.character} aria-hidden>
      <picture className={s.photo}>
        <Image
          className={s.image}
          src={photo}
          width={307.33}
          height={255.89}
          alt={name}
          layout="responsive"
        />
      </picture>
      <Field className={s.about}>
        <Field className={s.information} spacing="tiny">
          <Icon.Identification size={17} />
          <Text tag="span">{name}</Text>
        </Field>
        <Field className={s.information} spacing="tiny">
          <Icon.User size={17} />
          <Text tag="span">{gender}</Text>
        </Field>
        <Field className={s.information}>
          <Icon.Globe size={17} />
          <Text tag="span">{location}</Text>
        </Field>
      </Field>
    </article>
  );
};

export default Character;
