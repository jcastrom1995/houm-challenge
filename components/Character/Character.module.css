.character {
    border: 1px solid var(--colors-gray-300);
    border-radius: var(--border-radius-semi);
    box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.1), 0px 1px 2px rgba(0, 0, 0, 0.06);
}

.about {
    border-top: 1px solid var(--colors-gray-300);
    padding: 0.75rem;
}

.information {
    align-items: center;
    display: flex;
    justify-content: space-between;
}

.information svg {
    color: var(--colors-gray-600);
}

.photo {
    position: relative;
    width: 100%;
    height: auto;
}
