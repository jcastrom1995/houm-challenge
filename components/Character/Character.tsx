import { ReactElement } from "react";
import Image from "next/image";
import { Text, Field } from "ui";
import * as Icon from "Icons";
import s from "./Character.module.css";

interface CharacterI {
  photo: string;
  name: string;
  gender: string;
  location: string;
}

const Character = ({
  photo,
  name,
  gender,
  location,
}: CharacterI): ReactElement => {
  return (
    <Field className={s.character}>
      <picture className={s.photo}>
        <Image src={photo} alt={name} width={250} height={250} />
      </picture>
      <Field className={s.about}>
        <Field className={s.information} spacing="tiny">
          <Icon.Identification size={20} />
          <Text tag="span">{name}</Text>
        </Field>
        <Field className={s.information} spacing="tiny">
          <Icon.User size={20} />
          <Text tag="span">{gender}</Text>
        </Field>
        <Field className={s.information}>
          <Icon.Globe size={20} />
          <Text tag="span">{location}</Text>
        </Field>
      </Field>
    </Field>
  );
};

export default Character;
