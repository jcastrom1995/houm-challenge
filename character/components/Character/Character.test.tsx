import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import Character from "./Character";

describe("Home", () => {
  it("renders a heading", () => {
    render(
      <Character
        photo="https://rickandmortyapi.com/api/character/avatar/361.jpeg"
        name="Javier Castro"
        gender="Male"
        location="San Miguel"
        onSelect={() => jest.fn()}
      />
    );

    const name = screen.getByText("Javier Castro");

    expect(name).toBeInTheDocument();
  });
});
