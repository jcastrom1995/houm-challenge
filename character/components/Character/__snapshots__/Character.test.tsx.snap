// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading 1`] = `
<div>
  <article
    class="character"
  >
    <picture
      class="photo"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 83.26229134806235% 0px 0px 0px;"
        />
        <img
          alt="Javier Castro"
          class="image"
          data-nimg="responsive"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </picture>
    <div
      class="about"
    >
      <div
        class="spacing-tiny information"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          height="17"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="17"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 6H5a2 2 0 00-2 2v9a2 2 0 002 2h14a2 2 0 002-2V8a2 2 0 00-2-2h-5m-4 0V5a2 2 0 114 0v1m-4 0a2 2 0 104 0m-5 8a2 2 0 100-4 2 2 0 000 4zm0 0c1.306 0 2.417.835 2.83 2M9 14a3.001 3.001 0 00-2.83 2M15 11h3m-3 4h2"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span
          class="text"
        >
          Javier Castro
        </span>
      </div>
      <div
        class="spacing-tiny information"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          height="17"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="17"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span
          class="text"
        >
          Male
        </span>
      </div>
      <div
        class="information"
      >
        <svg
          class="h-6 w-6"
          fill="none"
          height="17"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="17"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.055 11H5a2 2 0 012 2v1a2 2 0 002 2 2 2 0 012 2v2.945M8 3.935V5.5A2.5 2.5 0 0010.5 8h.5a2 2 0 012 2 2 2 0 104 0 2 2 0 012-2h1.064M15 20.488V18a2 2 0 012-2h3.064M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        <span
          class="text"
        >
          San Miguel
        </span>
      </div>
    </div>
  </article>
</div>
`;
